pygmo/_py_problems.py

```diff
@@ -265,3 +265,158 @@ def get_decorator(self, fname):
                 "The input parameter 'fname' must be a string, but it is of type '{}' instead.".format(type(fname)))
         from copy import deepcopy
         return deepcopy(self._decors.get(fname))
+
+
+class constant_arguments():
+    """Meta problem that sets some arguments of the original problem to constants
+
+    .. versionadded:: 2.19
+
+    If good values for some of the dimensions of a problem are known, this
+    wrapper allows to reduce the dimensions of the search space and is an alternative
+    to restricting a value using identical lower and upper bounds.
+
+    We can construct an instance of this problem by passing the original problem,
+    the list of fixed arguments and a list of boolean flags, denoting for each argument
+    of the original dimensions whether it is fixed or not:
+
+    >>> from pygmo import constant_arguments, problem, rosenbrock
+    >>> cprob = problem(constant_arguments(rosenbrock(dim=3), fixed_arguments=[1, None, None]))
+
+    We now see that the new problem has two dimensions, since the original problem had three and we fixed one:
+
+    >>> cprob.get_nx()
+    2
+
+    """
+
+    def __init__(self, prob, fixed_arguments: List[Union[float, None]]):
+        """
+        Args:
+
+           prob: a :class:`~pygmo.problem` or a user-defined problem, either C++ or Python (if
+              *prob* is :data:`None`, a :class:`~pygmo.null_problem` will be used in its stead)
+           fixed_arguments: a list of values, one for each dimension of the wrapped problem.
+               Each value should be either a float, if the argument should be fixed to this value,
+               or None, if it should remain free
+
+        Raises:
+
+           ValueError: if the lengths of fixed_arguments differs from the number of dimensions of the wrapped problem
+           ValueError: if any of the fixed arguments violate the bounds of the wrapped problem
+           ValueError: if a problem with nix() > 0 is passed
+           unspecified: any exception thrown by the constructor of :class:`~pygmo.problem` or the deep copy
+              of *prob*
+        """
+
+        from . import problem, null_problem
+        from copy import deepcopy
+        if prob is None:
+            prob = null_problem()
+        if type(prob) == problem:
+            # If prob is a pygmo problem, we will make a copy
+            # and store it. The copy is to ensure consistent behaviour
+            # with the other meta problems and with the constructor
+            # from a UDP (which will end up making a deep copy of
+            # the input object).
+            self._problem = deepcopy(prob)
+        else:
+            # Otherwise, we attempt to create a problem from it. This will
+            # work if prob is an exposed C++ problem or a Python UDP.
+            self._problem = problem(prob)
+
+        minBound, maxBound = self._problem.get_bounds()
+
+        inner_dim = len(minBound)
```

Comment:
I am 99% sure that you cannot construct a ``pygmo.problem`` if inconsistent sizes are detected in the UDP's lower/upper bounds, so I believe this check is unnecessary.

Also, you can get the problem dimension directly from the ``pygmo.problem`` object, without querying for the size of the bounds.